app.py

```diff
@@ -185,17 +186,37 @@ def set_frame_range(self, engine, in_frame, out_frame):
 
         elif engine == "tk-nuke":
             import nuke
-
+            root = nuke.root()
+            lock_range = root["lock_range"]
             # unlock
-            locked = nuke.root()["lock_range"].value()
+            locked = lock_range.value()
+            change_msg = 'from %s-%s to %s-%s' % (prev_in, prev_out, in_frame, out_frame)
             if locked:
-                nuke.root()["lock_range"].setValue(False)
+                ret = nuke.ask("Frame range is currently locked,\n"
+                               "OK to update frame range %s ?" % change_msg)
+                if not ret:
+                    return
+            else:
+                p = nuke.Panel("Shall lock ?")
+                p.addMultilineTextInput("info",
+                                        "Frame range is going to be updated \n\n"
+                                        "%s,\n\n"
+                                        "please choose what to do.." % change_msg)
+                p.addButton('Reject')  # 0
+                p.addButton('Accept')  # 1
+                p.addButton('Accept but then LOCK the frame range')  # 2
+                ret = p.show()
+                if ret == 0:  # reject
+                    return
+                if ret == 2:  # accept and lock
+                    locked = True
+            lock_range.setValue(False)
             # set values
-            nuke.root()["first_frame"].setValue(in_frame)
-            nuke.root()["last_frame"].setValue(out_frame)
+            root["first_frame"].setValue(in_frame)
```

Comment:
As mentioned upper in the PR, I would prefer if we avoid unnecessary changes when possible. Same for line 26 and 219.